installer/tue-install-impl.bash

```diff
@@ -806,17 +1036,44 @@ TUE_INSTALL_STATE_DIR=$TUE_INSTALL_GENERAL_STATE_DIR/$stamp
 mkdir -p "$TUE_INSTALL_STATE_DIR"
 
 TUE_INSTALL_GIT_PULL_Q=()
+TUE_INSTALL_HG_PULL_Q=()
 
 TUE_INSTALL_SYSTEMS=
 TUE_INSTALL_PPA=
-TUE_INSTALL_PIPS=
+TUE_INSTALL_PIP2S=
 TUE_INSTALL_PIP3S=
 TUE_INSTALL_SNAPS=
 
 TUE_INSTALL_WARNINGS=
 TUE_INSTALL_INFOS=
 
+# Make sure tools used by this installer are installed
+tue-install-system-now git python-pip python3-pip
+
+tue-install-pip2-now mercurial
+
+tue-install-pip3-now distro PyYAML
+
+# Mercurial config extension to write configs from cli
+hgcfg_folder="$HOME"/src/hgcfg
+hgcfg_pulled=/tmp/hgcfg_pulled
+if [ ! -f "$hgcfg_pulled" ]
+then
+    parent_target=TUE_INSTALL_CURRENT_TARGET
+    TUE_INSTALL_CURRENT_TARGET="hgcfg"
+    tue-install-git "https://github.com/tue-robotics/hgconfig.git" "$hgcfg_folder"
```

Comment:
I don't like this. If possible move this to the targets repo and try to generalize this entire if block by using default config if none exists.

Reply:
This hg extension is required to edit the config in the command line. By default, hg opens an editor to edit the config.

The config contains the url, which we want to change if different.

Reply:
BTW we could move this inside tue-install-hg

Reply:
I still don't see where exactly this hgconfig is required for using `tue-install-hg`. All the commands in `tue-install-hg` are executable without the requirement of any custom config. Can you pin point the exact line where this configuration is needed to execute `tue-install-hg`?

Reply:
https://github.com/tue-robotics/tue-env/pull/453/files#diff-27df6d20d171684702149fd4165c21e7R320, here we change the path. Without the extension. you can get items from the configs. But to set them from command line, without an editor, the extension is needed.